### PR TITLE
Various 2021 05 22

### PR DESCRIPTION
**Pull request by jaredbeck**
A potential security vulnerability, network timeouts, a class extraction, replacing a gem with plain rails, and some general linting.

### CODE REVIEW

File: config/locales/en.yml [hunk 1]
```diff
@@ -101,6 +101,26 @@ en:
...
+        game analysis, continuous self-paired games, and all kinds of go-related
+        activities from morning to midnight. Come for the go. Come for the
+        camaraderie of old friends. Whatever your reason, we are looking forward
+        to seeing you there.
+      volunteers: >
+        From **tournament directors** to the people who volunteer **only a few
+        hours at the front desk**, we need your help.  The Go Congress would not
+        be possible without the dedication of dozens of volunteers.  Please
+        email the congress directors to volunteer at this year’s congress.
+        Thanks!
```
Comment (neagle) [inline config/locales/en.yml:123]: Nice! I really like this reorganization.

File: app/services/aga/mm/build_attendee.rb [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+          end
+        rescue OpenURI::HTTPError => e
+          # No worries! We just won't pre-fill any values.
+          ::Rails.logger.error(e)
+        end
+        attendee
+      end
+    end
+  end
+end
```
Comment (neagle) [inline app/services/aga/mm/build_attendee.rb:67]: I love it. When I create new features for the site, I do my best to find existing patterns and emulate them when possible, but often if I'm in relatively fresh territory, I just have to try for an approach that works. Having you provide a refactor that optimizes what I've done and helps provide a best-practice for the future is great.

**Comment by neagle** [comment]
Great PR to read through and learn from: thanks for the security fixes and for the improvements on how attendees are built with data from the AGA Membership Manager.

**Comment by jaredbeck** [comment]
Thanks for the kind words, Nate. Glad you like it.

**Pull request merged**